lib/Backends/OpenCL/OpenCLDeviceManager.cpp

```diff
@@ -64,15 +64,23 @@ OpenCLDeviceManager::OpenCLDeviceManager(std::unique_ptr<DeviceConfig> config)
 
 llvm::Error OpenCLDeviceManager::init() {
   // The OpenCl Backend defines three command line options: doProfile, deviceId,
-  // and platformId. If an OpenCLDeviceConfig is not provided we use the CL
+  // and platformId. If the parameter is not provided we use the CL
   // options from the OpenCl Backend.
 
-  if (config_ && config_->getBackendKind() == BackendKind::OpenCL) {
-    OpenCLDeviceConfig *config =
-        static_cast<OpenCLDeviceConfig *>(config_.get());
-    clDeviceId = config->deviceId;
-    clPlatformId = config->platformId;
-    clDoProfile = config->doProfile;
+  // Check if parameters are in map.
+  if (config_) {
+    auto it = config_->parameters.find("deviceId");
+    if (it != config_->parameters.end()) {
+      clDeviceId = std::stoi(it->second);
```

Comment:
What happens here if the input string is formatted incorrectly, and there's no int? Should we return an error?